src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonfwJavaProfile.java

```diff
@@ -0,0 +1,114 @@
+package com.devonfw.ide.sonarqube.common.impl;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition;
+import org.sonar.plugins.java.Java;
+import org.sonarsource.api.sonarlint.SonarLintSide;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * This class creates a quality profile containing the rules of this plugin plus additional rules from external repos.
+ */
+@SonarLintSide
+public class DevonfwJavaProfile implements BuiltInQualityProfilesDefinition {
+
+  private static final Set<String> FORBIDDEN_RULE_KEYS = new HashSet<>(Arrays.asList(
+      /* squid repo */
+      "S2076", "S2078", "S3318", "S2070", "S4142",
+      /* findbugs repo */
+      "VA_FORMAT_STRING_ARG_MISMATCH", "VA_FORMAT_STRING_MISSING_ARGUMENT", "VA_FORMAT_STRING_EXTRA_ARGUMENTS_PASSED",
+      "VA_FORMAT_STRING_BAD_CONVERSION_FROM_ARRAY", "VA_FORMAT_STRING_BAD_ARGUMENT",
+      "VA_FORMAT_STRING_BAD_CONVERSION_TO_BOOLEAN", "VA_FORMAT_STRING_NO_PREVIOUS_ARGUMENT", "VA_FORMAT_STRING_ILLEGAL",
+      "VA_FORMAT_STRING_BAD_CONVERSION", "VA_FORMAT_STRING_EXPECTED_MESSAGE_FORMAT_SUPPLIED",
+      /* fb-contrib repo */
+      "SPP_NULL_CHECK_ON_MAP_SUBSET_ACCESSOR", "SPP_NULL_CHECK_ON_OPTIONAL", "SPP_USE_CONTAINSKEY"));
+
+  private static final String DEVONFW_JAVA = "/com/devonfw/ide/sonarqube/common/rules/devon4j/devonfwJava.xml";
+
+  private static final Logger logger = Logger.getGlobal();
+
+  @Override
+  public void define(Context context) {
+
+    NewBuiltInQualityProfile devonfwJava = context.createBuiltInQualityProfile("devonfw Java", Java.KEY);
```

Comment:
So what happens if the server is restarted or our plugin gets updated?
Will this profile be updated every time? What if we remove a rule from the XML will it then persist in the profile forever?
What happens if you uninstall our plugin? Will this profile be removed or will it presist and potentially prevent the server from startup as it references non-existent rules?

Reply:
1. The profile is updated every time the server restarts, so when we create a new version where some rules were taken out of the XML, those rules will be deleted from the profile.
1. When the plugin is uninstalled, the profile stays on the server, only the rules from our plugin will be removed. This should not prevent the server from startup though, as the code that activates non-existent rules will not be on the server anymore.

Reply:
What might be problematic is that our quality profile is defined as "Built-In" so it can't be removed from the server once it has been created.